Fixed Grease Pencil nodes for latest Blender API changes.

3DSinghVFX opened the pull request:
I have fixed the Grease Pencil nodes for the latest [Blender 2.83 API](https://wiki.blender.org/wiki/Reference/Release_Notes/2.83/Grease_Pencil) changes.
Changes are:
1. Mainly related to the layer mask.
2. Small bug fixes.
3. Added _Hardeness_ attribute to GP Stroke.

3DSinghVFX commented:
@OmarEmaraDev I have added the _Vertex Color_ attribute to the _GP Stroke_ object.

We can add _Vertex Color_ input in these nodes which would be very handy,
![Screenshot from 2020-03-29 23-51-51](https://user-images.githubusercontent.com/30294746/77857058-10cf6e80-7219-11ea-8d5d-4514afa15667.png)

What do you think?

OmarEmaraDev commented:
@3DSinghVFX I don't see how vertex colors could be incorporated in those nodes, aside from the offset node. Can you elaborate?

3DSinghVFX commented:
> @3DSinghVFX I don't see how vertex colors could be incorporated in those nodes, aside from the offset node. Can you elaborate?

If we integrate the _Vertex Color_ in _Offset GP Stroke_  node then it is not needed for remaining nodes (especially the _GP Stroke From Spline_ and _Transform GP Stroke_ nodes). However, we can integrate _Vertex Color_ in _Replicate GP Stroke_ node, it will allow the replication of stroke with different vertex colors.

OmarEmaraDev commented:
I don't think we should add that to the Replicate node. The Offset node would be fine I guess.

3DSinghVFX commented:
> I don't think we should add that to the Replicate node. The Offset node would be fine I guess.

Okay. I'll make changes.

3DSinghVFX commented:
@OmarEmaraDev I have added the _offset color_ method with three basic color methods _ADD, MULTIPLY, MIX_ in `color/c_utils` file, and added the _Vertex Color_ offset option in the _Offset GP Stroke_ node.

3DSinghVFX commented:
Okay. I have done testing on the released Blender 2.83 build, it is working fine.

To get multiple layers by their name, we have to use multiple _GP Object Input_ nodes. Should we add _Get GP Layer From List_ node that will give a layer by name from the layer-list?
![Screenshot from 2020-06-04 11-36-00](https://user-images.githubusercontent.com/30294746/83729417-32a80000-a665-11ea-904e-4bd40ad34f50.png)

OmarEmaraDev commented on animation_nodes/nodes/gpencil/gp_layer_from_frames.py:
```diff
@@ -1,36 +1,71 @@
+        return GPLayer(layerName, frames, blendMode, opacity, useLights, tintColor, tintFactor, lineChange,
+                       passIndex, False, maskLayers)
+
+    def isEmptyLayer(self, layer):
```
You can move this to the GPLayer class as a class method, notice that it is also used in `set_gp_layer_attributes.py` Also, you can do something like this:
```
def isEmptyLayer(self):
    for frame in self.frames:
        if len(frame.strokes) > 0:
            return False
    return True
```

3DSinghVFX replied:
Okay.

3DSinghVFX replied:
Done. BTW why class method not as regular method?

OmarEmaraDev replied:
@3DSinghVFX My apologies, I meant a normal method.

3DSinghVFX replied:
Okay, I'll change that.

Should I removed this `if len(maskLayersIn) > 0`?. I added for the performance:
https://github.com/JacquesLucke/animation_nodes/blob/5c8005435cd226c5f3783899b7137bd0f4795fcf/animation_nodes/nodes/gpencil/gp_layer_from_frames.py#L50-L58

OmarEmaraDev replied:
@3DSinghVFX  It shouldn't make much of a difference. It is up to you.

3DSinghVFX replied:
Okay. I'll keep that.

OmarEmaraDev commented on animation_nodes/nodes/gpencil/gp_stroke_from_points.py:
```diff
@@ -1,39 +1,44 @@
         strengths = FloatList.fromValues(VirtualDoubleList.create(strengths, 1).materialize(amount))
         pressures = FloatList.fromValues(VirtualDoubleList.create(pressures, 1).materialize(amount))
         uvRotations = FloatList.fromValues(VirtualDoubleList.create(uvRotations, 0).materialize(amount))
+        vertexColors = ColorList.fromValues(VirtualColorList.create(vertexColors, Color((0, 0, 0, 0))).materialize(amount))
```
The output of `materialize` is already a ColorList, so no need for `ColorList.fromValues` here.

3DSinghVFX replied:
Okay.

3DSinghVFX replied:
Done.

3DSinghVFX commented:
> @3DSinghVFX Not in this patch. Lets talk about this later.

Okay.
